sale_automatic_workflow/models/sale_workflow_process.py

```diff
@@ -35,36 +35,41 @@ def _default_filter(self, xmlid):
         default='direct',
     )
     validate_order = fields.Boolean(string='Validate Order')
-    order_filter_domain = fields.Char(
+    order_filter_domain = fields.Text(
         string='Order Filter Domain',
-        default="[('state', '=', 'draft')]"
+        related='order_filter_id.domain',
```

Comment:
Looking at it harder, there is some inconsistency when there isn't something selected in the M2O. In light of that, I made the domains invisible when there isn't an associated filter record selected.

Regarding the readonly - unless we have a technical reason, I do not see the reason to require a more complex workflow just because we feel that a field is reference only. Interestingly enough, editing them through the related fields provided here is actually more convenient than editing them through the M2O because the `ir.filter` does not implement the `char_domain` widget.